src/matrix_auth_observer.py

```diff
@@ -112,7 +112,6 @@ def _get_matrix_auth_provider_data(
         Returns:
             MatrixAuthConfiguration instance.
         """
-        # future refactor victim: this is repeated with saml
         homeserver = (
             charm_state.synapse_config.public_baseurl
```

Comment:
I think this could be removed too, right? Since it is done in charm_state.

Reply:
(I mean, the check)

Reply:
nice catch, I removed this logic.